### PR TITLE
Builds on externals 4 2 stable

### PR DESCRIPTION
**Pull request by d-w-moore**
Contingent on  https://github.com/irods/externals/pull/49

### CODE REVIEW

**Comment by d-w-moore** [comment]
Addition JSON.HPP to top  level directory of this repo had to be reverted. The move to Clang6.0 and/or CMAKE 3.11 seems to be what brought the externals package version of `json.hpp` into include path visibility, causing build errors before said reversion.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Motivation
-The iRODS indexing capability provides a policy framework around both full text and metadata indexing for the purposes of enhanced data discovery.  Logical collections are annotated with metadata which indicates that any data objects or nested collections of data object should be indexed given a particular indexing technology, index type and index name.
+The iRODS indexing capability provides a policy framework around both full text and metadata idexing for the purposes of enhanced data discovery.  Logical collections are annotated with metadata which indicates that any data objects or nested collecitons of data object should be indexed given a particular indexing technology, index type and index name.
```
Comment (bh9) [inline README.md]: This seems to be just adding a pair of typos

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Where `<index name>` is an assumed existing index within the given technology, a
 
 The `<technology>` in the triple references the indexing technology, currently only suppored by `elasticsearch`.  This string is used to dynamically build the policy invocations when the indexing policy is triggered in order to delegate the operations to the appropriate rule engine plugin.
 
-The attribute is configurable within the `plugin_specific_configuration` for the indexing rule engine plugin.
+The attribute is configurable within the plugin_specific_configuration for the indexing rule engine plugin.
```
Comment (bh9) [inline README.md]: Personally, I find the formatting more useful

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR) #CPACK_DEBIAN_<COMPONENT>_PACKAGE_NAME
 
 
-find_package(IRODS 4.2.6 EXACT REQUIRED)
+find_package(IRODS 4.2.8 EXACT REQUIRED)
 set(IRODS_PLUGIN_VERSION_MAJOR 2)
 set(IRODS_PLUGIN_VERSION_MINOR 5)
```
Comment (bh9) [inline CMakeLists.txt]: Should this not also be updated in line with irods minor version?

File: README.md (hunk 1 above)
Comment (trel) [reply README.md]: agreed

File: README.md (hunk 2 above)
Comment (trel) [reply README.md]: agreed

File: CMakeLists.txt (hunk 3 above)
Comment (trel) [reply CMakeLists.txt]: needs to be converted to use the four digit version numbers.  can discuss whether 4.2.7.0 is best/possible before 4.2.8.0.

**Comment by d-w-moore** [comment]
Leaving room for later bump to 4.2.8.0

**Comment by d-w-moore** [comment]
tested on   centos 7 to vet lib64 change in 4-2-stable with indexing plugin.
However, please do not merge until 4.2.7 pre-release of indexing is tested

**Comment by d-w-moore** [comment]
This pull request is tested and ready for merge.

**Comment by trel** [comment]
squash?  i see five commits.

**Comment by d-w-moore** [comment]
At best, we could meld these two, because the latter of them is the reference point for the 4.2.7-compatible build of indexing:

```
[20] policy trigger AVU units now "elasticsearch"
favor local json.hpp, omit CMake externals JSON ref (4.2.7)
```

Then of course, we could squash the following three together into one, if that's preferable
```
remove drop-in "json.hpp", use externals
bump version to 4.2.8
[#23] elasticlient consortium number bump 0->1
 ```

**Comment by trel** [comment]
Yes, let's squash down to two.    The first will be the candidate SHA for a 4.2.7 tag.

**Comment by d-w-moore** [comment]
Commits now properly #'d

**Pull request merged**